scripts/quests/otherAreas/The_Sand_Charm.lua

```diff
@@ -0,0 +1,144 @@
+                        return quest:event(125)
+                    elseif quest:getVar(player, "status") == 2 then
+                        return quest:event(124)
+                    end
+                end,
+            },
+            ['Zexu'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, "status") == 1 then
+                        return quest:event(123)
+                    end
+                end,
+            },
+            ['Celestina'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, "status") == 3 then
+                        return quest:event(126, xi.items.SAND_CHARM)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [123] = function(player, csid, option, npc)
+                    quest:incrementVar(player, "status", 1)
+                end,
+
+                [124] = function(player, csid, option, npc)
+                    quest:incrementVar(player, "status", 1)
+                end,
+
+                [125] = function(player, csid, option, npc)
+                    quest:incrementVar(player, "status", 1)
+                end,
+
+                [126] = function(player, csid, option, npc)
+                    if option == 70 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.MHAURA] =
+        {
+
+            ['Celestina'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHas(trade, xi.items.SAND_CHARM) then
+                        return quest:event(127, 0, xi.items.SAND_CHARM)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [127] = function(player, csid, option, npc)
+                    if player:hasKeyItem(xi.keyItem.MAP_OF_BOSTAUNIEUX_OUBLIETTE) then
+                        player:addExp(2000)
+                        npcUtil.giveCurrency(player, 'gil', 2000)
+                    else
+                        npcUtil.giveKeyItem(player, xi.keyItem.MAP_OF_BOSTAUNIEUX_OUBLIETTE)
+                    end
+
+                    player:setCharVar("blandineThanks", 1)
+                    player:tradeComplete()
+                    quest:complete(player)
+                end,
+            },
+        },
+    },
+    {
+        {
+            check = function(player, status, vars)
+                return status == QUEST_COMPLETED
+            end,
+
+            [xi.zone.MHAURA] =
+            {
+                ['Blandine'] =
+                {
+                    onTrigger = function(player, npc)
+                        if player:getCharVar("blandineThanks") == 1 then
```

Comment:
I believe you can still use quest vars here :) Not that it makes that big of a difference.